rubix-bookkeeper/src/main/java/com/qubole/rubix/bookkeeper/BookKeeper.java

```diff
@@ -319,6 +322,34 @@ public boolean readData(String remotePath, long offset, int length, long fileSiz
     }
   }
 
+  @Override
+  public FileInfo getFileInfo(String remotePath) throws TException
+  {
+    if (CacheConfig.isFileInvalidationEnabled(conf)) {
+      try {
+        Path path = new Path(remotePath);
+        FileSystem fs = path.getFileSystem(conf);
+        FileStatus status = fs.getFileStatus(path);
+        FileInfo info = new FileInfo(status.getLen(), status.getModificationTime());
+        return info;
+      }
+      catch (Exception e) {
+        log.error(String.format("Could not fetch FileStatus from remote file system for %s : %s", remotePath, Throwables.getStackTraceAsString(e)));
+      }
+    }
+    else {
+      try {
+        return fileInfoCache.get(remotePath, new FetchFileInfoCallable(remotePath, conf));
```

Comment:
Curious why you chose the Callable paradigm instead of the CacheLoader/Load fn. paradigm. Is it because of the conf variable ? I suggest you comment why you have chosen a different paradigm to the rest of the caches in this project.

Reply:
I chose to implement through Callable because the other cache (FileMetadtaCache) was using the same. Read when to use Callable and CacheLoader/Load fn. Changed the implementation to CacheLoader/Load fn.

+1 for the review comment.